File: pbcommonwcl/projects/pbandi/common-lib/src/lib/cronoprogramma-fasi/commons/dto/cronoprogramma-list-fasi-item-allegati.ts
/*
 * Copyright Regione Piemonte - 2024 
 * SPDX-License-Identifier: EUPL-1.2 
*/

import { AllegatiCronoprogrammaFasi } from "./allegati-cronoprogramma-fasi";
import { CronoprogrammaFasiItem } from "./cronoprogramma-fasi-item";

export class CronoprogrammaListFasiItemAllegati {

  fasiItemList: Array<CronoprogrammaFasiItem>
  allegatiList: Array<AllegatiCronoprogrammaFasi>

  constructor() {
    this.fasiItemList = new Array<CronoprogrammaFasiItem>()
    this.allegatiList = new Array<AllegatiCronoprogrammaFasi>()
  }

  pushItem(item: CronoprogrammaFasiItem) {
    this.fasiItemList.push(item)
  }

  pushAllegati(allegato: AllegatiCronoprogrammaFasi) {
    this.allegatiList.push(allegato);
  }

}

File: pbservizit/pbservizit-jar/src/main/java/it/csi/pbandi/pbservizit/dto/cronoprogrammaFasi/CronoprogrammaListFasiItemAllegati.java
/*******************************************************************************
* Copyright Regione Piemonte - 2024
* SPDX-License-Identifier: EUPL-1.2
******************************************************************************/

package it.csi.pbandi.pbservizit.dto.cronoprogrammaFasi;

import java.util.LinkedList;
import java.util.List;

public class CronoprogrammaListFasiItemAllegati implements java.io.Serializable {	
	
	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	
	private List<CronoprogrammaFasiItem> fasiItemList;
	private List<AllegatiCronoprogrammaFasi> allegatiList;
	
	
	
	
	public CronoprogrammaListFasiItemAllegati() {		
		this.fasiItemList = new LinkedList<CronoprogrammaFasiItem>();
		this.allegatiList = new LinkedList<AllegatiCronoprogrammaFasi>();
	}
	
	
	public List<CronoprogrammaFasiItem> getFasiItemList() {
		return fasiItemList;
	}
	public void setFasiItemList(List<CronoprogrammaFasiItem> fasiItemList) {
		this.fasiItemList = fasiItemList;
	}
	public List<AllegatiCronoprogrammaFasi> getAllegatiList() {
		return allegatiList;
	}
	public void setAllegatiList(List<AllegatiCronoprogrammaFasi> allegatiList) {
		this.allegatiList = allegatiList;
	}
		
	public boolean addItem (CronoprogrammaFasiItem allegatiList) {
		return fasiItemList.add(allegatiList);
	}
	
	

	
	
	
}
